components/team_controller/team_controller.tsx

```diff
@@ -0,0 +1,295 @@
+            if (teamByName && teamByName.delete_at === 0) {
+                const {error} = await props.addUserToTeam(teamByName.id, props.currentUser && props.currentUser.id);
+                if (error) {
+                    return null;
+                }
+
+                // User added to the new team, now attempt to initialize the team
+                if (firstLoad) {
+                    LocalStorageStore.setTeamIdJoinedOnLoad(teamByName.id);
+                }
+                return teamByName;
+            }
+            return null;
+        } catch (error) {
+            return null;
+        }
+    }
+
+    async function initTeam(team: Team) {
+        // If current team is set, then this is not first load
+        // The first load action pulls team unreads
+        props.getMyTeamUnreads(props.collapsedThreads);
+        props.selectTeam(team);
+        props.setPreviousTeamId(team.id);
+
+        if (props.currentUser && isGuest(props.currentUser.roles)) {
+            setIsFetchingChannels(true);
+        }
+
+        await props.fetchMyChannelsAndMembers(team.id);
+        setIsFetchingChannels(false);
+
+        props.loadStatusesForChannelAndSidebar();
+
+        if (props.license &&
+            props.license.IsLicensed === 'true' &&
+            (props.license.LDAPGroups === 'true' || props.isCustomGroupsEnabled)) {
+            if (props.currentUser) {
+                props.getGroupsByUserIdPaginated(props.currentUser.id, false, 0, 60, true);
+            }
+
+            if (props.license.LDAPGroups === 'true') {
+                props.getAllGroupsAssociatedToChannelsInTeam(team.id, true);
+            }
+
+            if (team.group_constrained && props.license.LDAPGroups === 'true') {
+                props.getAllGroupsAssociatedToTeam(team.id, true);
+            } else {
+                props.getGroups(false, 0, 60);
+            }
+        }
+    }
+
+    async function joinTeamAndInit() {
+        try {
+            const joinedTeam = await joinTeam();
+
+            if (joinedTeam) {
+                initTeam(joinedTeam);
+                setTeam(joinedTeam);
+            } else {
+                throw new Error('Unable to join team');
+            }
+        } catch (error) {
+            history.push('/error?type=team_not_found');
+        }
+    }
+
+    // Effect to run when url for team changes
+    useEffect(() => {
+        const team = getTeamFromTeamParam(props.teamsList, teamNameParam);
+
+        if (team) {
+            // User belongs to team found in url
+            initTeam(team);
+            setTeam(team);
+
+            // Prevents the RHS from closing when clicking on a global permalink.
+            emitCloseRightHandSide();
+        } else {
+            // User is not a member of the team found in the url
+            // attempt to join the team
+            joinTeamAndInit();
+        }
+    }, [teamNameParam]);
+
+    if (props.mfaRequired) {
+        history.push('/mfa/setup');
```

Comment:
Strictly speaking, we should be waiting until a `useEffect` to trigger this change, although we did it in the contructor before which was also incorrect. I don't know if having this here has any negative side effects though or if it's just not a great practice like how we did that in the constructor before

Reply:
this props comes from the initial load of config and license 🤔  it doesnt have dependency on the use effect of this component